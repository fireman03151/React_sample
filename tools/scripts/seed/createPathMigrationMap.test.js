/* global describe expect */
const { isPlainObject } = require('lodash');

const { createPathMigrationMap } = require('./createPathMigrationMap');
const mockCurriculum = require('./__mocks__/curriculum.json');

describe('createPathMigrationMap', () => {
  const pathMap = createPathMigrationMap(mockCurriculum);

  it('is a function', () => {
    expect(typeof createPathMigrationMap).toEqual('function');
  });

  it('returns an object', () => {
    expect(isPlainObject(pathMap)).toBe(true);
  });

  it('maps a challenge title to the correct uri slug', () => {
    expect.assertions(3);
    const slugOne = '/learn/super-block-b/block-one/challenge-ten';
    const slugTwo = '/learn/super-block-a/block-two/challenge-five';
    const slugThree = '/learn/super-block-a/block-one/challenge-three';

    expect(pathMap['challenge-ten']).toEqual(slugOne);
    expect(pathMap['challenge-five']).toEqual(slugTwo);
    expect(pathMap['challenge-three']).toEqual(slugThree);
  });

  it('does not add uri migrations for private challenges', () => {
    expect(pathMap['challenge-nine']).toBeUndefined();
  });

  it('output snapshot', () => {
    expect(createPathMigrationMap(mockCurriculum)).toMatchSnapshot();
  });
});
