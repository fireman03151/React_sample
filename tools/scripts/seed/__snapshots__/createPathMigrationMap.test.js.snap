// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createPathMigrationMap output snapshot 1`] = `
Object {
  "challenge-eight": "/learn/super-block-a/block-two/challenge-eight",
  "challenge-eleven": "/learn/super-block-b/block-one/challenge-eleven",
  "challenge-five": "/learn/super-block-a/block-two/challenge-five",
  "challenge-four": "/learn/super-block-a/block-one/challenge-four",
  "challenge-one": "/learn/super-block-a/block-one/challenge-one",
  "challenge-seven": "/learn/super-block-a/block-two/challenge-seven",
  "challenge-six": "/learn/super-block-a/block-two/challenge-six",
  "challenge-ten": "/learn/super-block-b/block-one/challenge-ten",
  "challenge-three": "/learn/super-block-a/block-one/challenge-three",
  "challenge-twelve": "/learn/super-block-b/block-one/challenge-twelve",
  "challenge-two": "/learn/super-block-a/block-one/challenge-two",
}
`;
